### ISSUE TITLE
how to run the peoject?

### ISSUE THREAD

**Issue by Forrest-Z** [josiasalexandre/hybrid_astar#1]
hi,do you have wiki to this project ? thank you

**Comment by josiasalexandre** [comment]
Hi,
It's an old project and it was meant to run in the outdated CARMEN framework.

Let me know what is your project and how I can help you.

Josias

**Comment by Forrest-Z** [comment]
Hi,
   I building the project without CARMEN successful,.When i put the map data to the project ,the Voronoi Diagram feel like wrong,I save the Voronoi Diagram image,the color is blue,do you have any idea with it ?

**Comment by josiasalexandre** [comment]
The voronoy diagram uses a RDDF message in order to build the obstacle distance map and the voronoy diagram around the desired path. See this video: https://www.youtube.com/watch?v=EcyTxEO0qKI

Let me know please what is your project.

**Comment by Forrest-Z** [comment]
I want to implement hybrid astar path planner for car-like robot , I saw  the hybrid astar path planner implement by standford , it's amazin work . When I saw you project , I want to do some work base you project. Do you know any path planner for car-like robot ? thank you.

**Comment by Forrest-Z** [comment]
Thanks

**Comment by xilinnancheng** [comment]
I have a question whther the gradient of curvature term is right.
Maybe we can sync with it.

**Comment by josiasalexandre** [comment]
Hi!

Let me know please your consideration. Feel free to contribute.

What do you think about creating a new project together and replace the optimizer and the vector library?

**Comment by xilinnancheng** [comment]
The derivative of k in the paper has same wrong sign. 
And p1 and p2 are relative to delta_xi and delta_xi+1 rather than xi and xi+1.  
Maybe you can try to calculate it by yourself.

**Comment by xilinnancheng** [comment]
I don't know how to run your project. Can you make your project base on ROS?

**Comment by josiasalexandre** [comment]
Hi! It would be great to port this project to ROS? Can we make this effort joinlty?

I would take another look at the derivatives and I'll get back to you.

**Comment by xilinnancheng** [comment]
Yes,let us work it together!

**Comment by josiasalexandre** [comment]
Great!

Are you using ROS2?

Let me know please your context and your project! I would be very glad to help you.

**Comment by josiasalexandre** [comment]
My email is josiasalexandre@gmail.com and we can use another channel if you like.

**Comment by xilinnancheng** [comment]
I work on my corp's environment so i can't share the project online.
Maybe I can create my own simple project on ROS1 to test it.

**Comment by josiasalexandre** [comment]
I see. I'm a curious person. Just describe your project, what are you doing, if possible.

I'll start a ROS1 project, let me know the version and I'll setup my environment. I'm using ROS2 in my projects. I'm finishing my masters and I'm currently starting my startup. We are focused in robotics related stuff.

**Comment by xilinnancheng** [comment]
OK, I'm a self-driving engineer and work on freespace planning which we want to use it to handle complex scenes. I'm also finishing my masters.

**Comment by josiasalexandre** [comment]
That's great!

So let's start! Let me know wich ROS version and I'll share a new project with you.

**Comment by josiasalexandre** [comment]
I want to move to Eigen library and use another Conjugate Gradient Optimizer. What do you think?

**Comment by xilinnancheng** [comment]
Kinetic is OK. Thanks!

**Comment by xilinnancheng** [comment]
No problem

**Comment by josiasalexandre** [comment]
This project makes usage of an online Voronoy diagram wich is very expensive in my implementation (http://www2.informatik.uni-freiburg.de/~lau/paper/lau10iros.pdf). Are you going to pre-compute the voronoy diagram and serve it through the map server?
I build this project long time ago, for a given undergraduate course and at that time we had to care of moving objects so an offline voronoy diagram was not appropriate. This Voronoy diagram also have a convenient way to retrieve the nearest obstacle given a grid map cell.

Maybe it's possible to update the Voronoy Diagram to GPU in a different approach or use another ready to go implementation. Do you know another options?

**Comment by josiasalexandre** [comment]
Which license is appropriate to your company?

**Comment by xilinnancheng** [comment]
How much time does the voronoi-diagram consumes? I also use the same method in my own project.
Maybe the method in https://github.com/ethz-asl/mav_voxblox_planning will give you some ideas. This method is applied in 3D environment.

**Comment by xilinnancheng** [comment]
MIT license is OK. I just want to verify this cg-optimizer method can work well.

**Comment by josiasalexandre** [comment]
At that time, a 150 x 150m map with 0.2m grid cell took around 1s, but our autonomous vehicle design required at least 10Hz.
Do you use voxblox in your current work? It looks great and it would be very nice to integrate in this project. Send me your impressions  or review, if you have experienced voxblox before.

**Comment by xilinnancheng** [comment]
No, I don't use voxblox currently.  I think the voxblox has same idea with (http://www2.informatik.uni-freiburg.de/~lau/paper/lau10iros.pdf). I just investigated the method which the voxblox use it to compare different obstacle.

**Comment by xilinnancheng** [comment]
First setup voronoi-diagram takes 1s. But I think the time when replan process take may decrease. Because only partial part of environment updates.

**Comment by josiasalexandre** [comment]
I have invited you to a new repo. I'll update the code there and you can already start a new ros project inside the repo and push your starting project. I'll keep inserting the code and including the updates.

We probably have very different timezones, so I'll take a brake now and return later. It's midnight over here.

If you want we can make a call thursday and I'll show you this old project already running in a Carmen environment.

Thanks!

**Comment by josiasalexandre** [comment]
I think I had a worse time in the first setup and 1s in the next updates. It was probably due to the low power pc and the unoptimized code.

**Comment by xilinnancheng** [comment]
OK,thanks!

**Comment by xilinnancheng** [comment]
Hi, i can't find your new repro. please ask me if you create it. Btw, I have found the paper with correct equation. https://sci-hub.se/10.1109/access.2018.2845448

**Comment by josiasalexandre** [comment]
https://github.com/josiasalexandre/ros_hybrid_astar/invitations

**Comment by xilinnancheng** [comment]
OK, Thanks.

**Comment by josiasalexandre** [comment]
Can you start your ROS1 project and push it to the new repo?

**Comment by josiasalexandre** [comment]
?

**Comment by josiasalexandre** [comment]
I have found a lot of issues in the smoother but I finally got the correct solution. It works now. As you mentioned, there was some error in the derivatives, including other issues indicated elsewere.

Currently, I have an working version running in ROS Dashing. I'll create another repo in the near future.
